index.js

```diff
@@ -454,5 +448,99 @@ module.exports = {
      */
     reset() {
         webpackConfig = new WebpackConfig(runtimeConfig);
-    }
+    },
+
+    /**
+     * Initialize the runtime environment.
+     *
+     * It can be used to directly manipulate the Encore API without
+     * executing the "./node_module/.bin/encore" utility.
+     *
+     * Encore.configureRuntimeEnvironment(
+     *     // Environment to use (dev, dev-server, production)
+     *     'dev-server',
+     *
+     *     // Same options you would use with the
+     *     // CLI utility with their name in
+     *     // camelCase.
+     *     {
+     *         https: true,
+     *         keepPublicPath: true
+     *     }
+     * )
+     *
+     * Be aware than using this method will also reset the current
+     * webpack configuration.
+     *
+     * @param {string} environment
+     * @param {object} options
+     * @returns {exports}
+     */
+    configureRuntimeEnvironment(environment, options = {}) {
+        runtimeConfig = parseRuntime(
+            Object.assign(
+                {},
+                require('yargs/yargs')([environment]).argv,
```

Comment:
I added that part to make sure that the first parameter passed to `parseRuntime` had the same structure than the one usually supplied by yargs, it doesn't actually keep reading the CLI options since we provide a fake `process.argv` (see https://github.com/yargs/yargs/blob/master/docs/api.md#api).

I did `options._ = [environment]` at first but switched to calling the API after realizing that `yargs` adds other things to the `argv` array that may be used later by `parseRuntime` (e.g.: `$0`)